ownCloudTests/Security/PasscodeTests.swift

```diff
@@ -65,4 +71,160 @@ class PasscodeTests: XCTestCase {
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number3Button")).perform(grey_tap())
+
+        // Asserts
+        EarlGrey.select(elementWithMatcher: grey_text("The entered codes are different")).assert(grey_sufficientlyVisible())
+        EarlGrey.select(elementWithMatcher: grey_text("Cancel")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("passcodeSwitchIdentifier")).assert(grey_switchWithOnState(false))
+
+        //Reset status
+        EarlGrey.select(elementWithMatcher: grey_text("ownCloud")).perform(grey_tap())
+    }
+
+    /*
+     * PASSED if: Passcode lock is correctly disabled in Settings view
+     */
+    func testDisablePasscode() {
+
+        // Prepare the simulator show the passcode
+        AppLockManager.shared.passcode = "1111"
+        AppLockManager.shared.lockEnabled = true
+
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("settingsBarButtonItem")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("passcodeSwitchIdentifier")).perform(grey_turnSwitchOn(false))
+
+        // Tap the number buttons first time
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+
+        // Asserts
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("passcodeSwitchIdentifier")).assert(grey_switchWithOnState(false))
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("lockFrequency")).assert(grey_notVisible())
+
+        //Reset status
+        EarlGrey.select(elementWithMatcher: grey_text("ownCloud")).perform(grey_tap())
+    }
+
+    /*
+     * PASSED if: Passcode lock keeps enabled in Settings view after cancelling
+     */
+    func testCancelDisablePasscode() {
+
+        // Prepare the simulator show the passcode
+        AppLockManager.shared.passcode = "1111"
+        AppLockManager.shared.lockEnabled = true
+
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("settingsBarButtonItem")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("passcodeSwitchIdentifier")).perform(grey_turnSwitchOn(false))
+
+        // Tap the number buttons first time
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_accessibilityID("number1Button")).perform(grey_tap())
+        EarlGrey.select(elementWithMatcher: grey_text("Cancel")).perform(grey_tap())
```

Comment:
You should use the text from the translated file because maybe the text change or the device for testing is in other language

Reply:
👍